### DISCUSSION TITLE
Custom cache composition and loadout

### DISCUSSION THREAD

**Discussion by AshTrayBY** [Vdauphin/HeartsAndMinds#1557] [Question and answer] [answered]
Hi,
want to create early '80 cache composition and loadout but afraid to break something in mission.
Would you be so king to show how to change array of random useful weapons to list of exact weapons
//Cache
    btc_cache_type = [
        //_allClassSorted select {
        //    _x isKindOf "ReammoBox_F" &&
        //    {getText(_cfgVehicles >> _x >> "model") isEqualTo "\A3\weapons_F\AmmoBoxes\AmmoBox_F"}
        //},
        ["Box_IED_Exp_F", "rhs_7ya37_AKM"]
    ];
    private _weapons_usefull = "true" configClasses (configFile >> "CfgWeapons") select {
        getNumber (_x >> 'type') isEqualTo 1 &&
        {getArray (_x >> 'magazines') isNotEqualTo []} &&
        {getNumber (_x >> 'scope') isEqualTo 2}
    };
    btc_cache_weapons_type = _weapons_usefull apply {(toLower getText (_x >> "model")) select [1]};

**Comment by Vdauphin** [comment, marked as answer]
Hello,
Not sure about what you wrote but basically:

```sqf
    //Cache
    btc_cache_type = [
        ["ammoboxClassname1", "ammoboxClassname2"],
        ["Land_PlasticCase_01_small_black_CBRN_F", "Land_PlasticCase_01_small_olive_CBRN_F", "Land_PlasticCase_01_small_CBRN_F"]
    ];
    private _weapons_usefull = ["weaponClassname1", "weaponClassname2"];
 ```
Replace with your custom ammo box classname and weapon


Cheers

**Comment by AshTrayBY** [comment]
Want to remove barrels from hideouts:
replacing     
//Hideout classname
    btc_type_campfire = ["MetalBarrel_burning_F"] + (_allClassSorted select {_x isKindOf "Land_Campfire_F"});
with
    btc_type_campfire = ["CampFire_burning_F"] + (_allClassSorted select {_x isKindOf "Land_Campfire_F"});
or
btc_type_campfire = ["Land_Campfire_F"];
caused mission not to start. what I'm doing wrong?

**Reply by Vdauphin** [reply]
Do not hesitate to create new discussion when facing an other problem
Looks good for me:
```sqf
btc_type_campfire = ["Land_Campfire_F"];
```
